Allow using a custom AndroidManifest.xml.in file

Earthwings opened the pull request:
To use a custom AndroidManifest.xml.in, pass it via the
PACKAGE_MANIFEST argument of add_qt_android_apk. If PACKAGE_MANIFEST is
not specified, AndroidManifest.xml.in from qt-android-cmake is used.

Earthwings committed:
Allow using a custom AndroidManifest.xml.in file

To use a custom AndroidManifest.xml.in, pass it via the
PACKAGE_MANIFEST argument of add_qt_android_apk. If PACKAGE_MANIFEST is
not specified, AndroidManifest.xml.in from qt-android-cmake is used.

LaurentGomila commented:
Since you can have a custom manifest in the PACKAGE_SOURCES, what does this new option adds, other than more confusion (I don't like having two ways to do the same thing)?

Earthwings committed:
Configure AndroidManifest.xml.in in PACKAGE_SOURCES, if given

If there is a file AndroidManifest.xml.in in PACKAGE_SOURCES, configure
it like the template file used when PACKAGE_SOURCES is not given.

Earthwings commented:
I'd like to run the manifest through configure_file(). For PACKAGE_SOURCES it currently expects a final AndroidManifest.xml.

I pushed another commit that reuses PACKAGE_SOURCES and checks if an AndroidManifest.xml.in exists inside. In that case it runs configure_file on it. What do you think about that approach?

LaurentGomila commented:
It seems like a rather specific usage. What's your use case?

Earthwings commented:
It takes care of inserting the required Qt libs automatically.

LaurentGomila commented:
This is done by the Qt tool, not CMake's `configure_file`.

Earthwings commented:
Indeed, I missed that. Makes this unneeded then.
